src/view.c

```diff
@@ -673,16 +674,15 @@ view_toggle_fullscreen(struct view *view)
 }
 
 void
-view_set_fullscreen(struct view *view, bool fullscreen,
-		struct wlr_output *wlr_output)
+view_set_fullscreen(struct view *view, bool fullscreen, struct output *output)
 {
 	assert(view);
-	if (fullscreen != !view->fullscreen) {
+	if (fullscreen == view->fullscreen) {
```

Comment:
That is so much better, can't count how many times i've had to stop here and verify that it was actually correct before.